### PR TITLE
Update Azure Secrets docs

### PR DESCRIPTION
**Pull request by kalafut**
Add coverage of application_object_id parameter.

### CODE REVIEW

File: website/source/api/secret/azure/index.html.md [hunk 1]
```diff
@@ -123,9 +124,12 @@ for more information about roles.
 
 ### Parameters
 
-- `azure_roles` (`string: <required>`) - List of Azure roles to be assigned to the generated service
+- `azure_roles` (`string: ""`) - List of Azure roles to be assigned to the generated service
    principal. The array must be in JSON format, properly escaped as a string. See [roles docs][roles]
    for details on role definition.
+- `application_object_id` (`string: ""`) - Application Object ID for an existing service principal that will
+   be used instead of creating dynamic service principals. If present, `azure_roles` will be ignored. See
+   [roles docs][roles] for details on role definition.
```
Comment (tyrannosaurus-becks) [inline website/source/api/secret/azure/index.html.md]: Is it intended for there to be two successive sets of brackets here?

File: website/source/docs/secrets/azure/index.html.md [hunk 2]
```diff
@@ -50,10 +54,16 @@ management tool.
     If you are running Vault inside an Azure VM with MSI enabled, `client_id` and
     `client_secret` may be omitted. For more information on authentication, see the [authentication](#authentication) section below.
 
-1. Configure a role. Roles determine the permissions that the service principal
-generated by Vault will have to Azure resources.
+1. Configure a role. A role may be set up with either an existing service principal, or
+a set of Azure roles that will be assigned to a dynamically created service principal.
+
+    To configure a role called "my-role" with an existing service principal:
```
Comment (tyrannosaurus-becks) [inline website/source/docs/secrets/azure/index.html.md]: Is it intended for this line to be indented? I also notice a bunch of other non-code lines are indented from here thru line 80. If it doesn't make any difference in how the text is displayed, then it's nothing to worry about, but just thought I'd ask.

File: website/source/docs/secrets/azure/index.html.md [hunk 3]
```diff
@@ -153,6 +172,27 @@ The following Azure roles and Azure Active Directory (AAD) permissions are requi
...
+completely decoupled from any other clients, is not subject to permission changes after
+issuance, and offers the best audit granularity.
+
+Access to some Azure services cannot be provided with the RBAC system, however. In these
+cases, an existing service principal can be set up with the necessary access, and Vault
+can create new passwords for this service principal. Any changes to the service principal
+permissions affect all clients. Furthermore, Azure does not provide any logging with
+regard to _which_ credential was used for an operation.
+
+An important limitation when using an existing service prinicpal is that Azure limits the
```
Comment (tyrannosaurus-becks) [inline website/source/docs/secrets/azure/index.html.md]: Typo: "principal".

File: website/source/api/secret/azure/index.html.md (hunk 1 above)
Comment (kalafut) [reply website/source/api/secret/azure/index.html.md]: Yep!

File: website/source/docs/secrets/azure/index.html.md (hunk 2 above)
Comment (kalafut) [reply website/source/docs/secrets/azure/index.html.md]: Updated.

**Pull request merged**